win32/tests/sync_win32_ut/sync_win32_ut.c

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdbool.h>
+#endif
+#include "windows.h"
+#include "azure_macro_utils/macro_utils.h"
+#include "testrunnerswitcher.h"
+
+#include "umock_c/umock_c.h"
+#include "umock_c/umocktypes_windows.h"
+
+#include "sync.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#define ENABLE_MOCKS
+#include "mock_sync.h"
+#undef ENABLE_MOCKS
+
+#ifdef __cplusplus
+}
+#endif
+
+BEGIN_TEST_SUITE(sync_win32_unittests)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+    ASSERT_ARE_EQUAL(int, 0, umock_c_init(on_umock_c_error));
+    ASSERT_ARE_EQUAL(int, 0, umocktypes_windows_register_types());
+}
+
+TEST_SUITE_CLEANUP(TestClassCleanup)
+{
+    umock_c_deinit();
+
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(f)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+
+    umock_c_reset_all_calls();
+}
+
+TEST_FUNCTION_CLEANUP(cleans)
+{
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+/*Tests_SRS_SYNC_WIN32_43_001: [ wait_on_address shall call WaitOnAddress from windows.h with address as Address, compare_address as CompareAddress, 4 as AddressSize and timeout_ms as dwMilliseconds. ]*/
+/*Tests_SRS_SYNC_WIN32_43_002: [ wait_on_address shall return the return value of WaitOnAddress ]*/
+TEST_FUNCTION(wait_on_address_calls_WaitOnAddress_successfully)
+{
+    ///arrange
+    volatile int32_t var;
+    int32_t val = INT32_MAX;
+    InterlockedExchange((volatile LONG*)&var, val);
+    uint32_t timeout = 1000;
+    STRICT_EXPECTED_CALL(mock_WaitOnAddress((volatile VOID*)&var, (PVOID)&val, (SIZE_T)4, (DWORD)timeout))
+        .SetReturn(true);
+
+    ///act
+    bool return_val = wait_on_address(&var, &val, timeout);
+
+    ///assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls(), "Actual calls differ from expected calls");
+    ASSERT_IS_TRUE(return_val, "Return value is incorrect.");

```

Comment:
>"Return value is incorrect." [](start = 31, length = 28)

Probably don't need the message on the assert unless you are adding some additional information for debugging/reading the test results.